### ISSUE TITLE
nixos-generate-config needs to use acpilight for the Framework Laptop

### ISSUE THREAD

**Issue by IreneKnapp** [NixOS/nixpkgs#171093]
### Describe the bug
On the Framework laptop, nixos-generate-config sets hardware.acpilight.enable to false (thus, excluding it from the generate config, as false is the default). This is incorrect, it needs to be true, as I will describe below.

This is causing no inconvenience to me personally, and I have no prior familiarity with the internals of nixos-generate-config, so I don't intend to work on it unless it turns to be an easy fix. I am reporting it because this is an increasingly popular laptop model for Linux users, so we really ought to make sure to handle it correctly.

### Steps To Reproduce
1. Boot into a nix system (an install image is fine) on the Framework Laptop (any sub-model).
2. Run nixos-generate-config.
3. In the hardware-configuration.nix which is produced, observe that there is no mention of hardware.acpilight.enable.
4. Complete the install based on the generated configuration.
5. Use the program "light", or any similar program, to adjust the backlight brightness.
6. Observe that the backlight brightness does not change.

### Expected behavior
The generated hardware-configuration.nix would set hardware.acpilight.enable = true. This would, among other things, install the xbacklight program, which does work correctly once installed.

### Screenshots
I have omitted this because screenshots don't tell you how bright the screen is. :)

### Additional context
In searching for advice on this problem, I found what appears to be some closely-related information. https://wiki.archlinux.org/title/Framework_Laptop#Changing_the_brightness_of_the_monitor_does_not_work

Specifically, according to that page, the relevant kernel modules on this laptop model produce a directory tree /sys/class/backlight/intel_backlight/ which resembles the other, non-acpilight way to do things. However, again per that page, this directory tree does not actually function in the way that non-acpilight tooling would like it to. It should be clear how this is likely to confuse the kind of hardware detection that nixos-generate-config does.

### Notify maintainers
@setser you're the listed maintainer of acpilight as a package, and neither nixos-generate-config nor the acpilight module have maintainers listed, so I'm @'ing you. Feel free to ignore if you're not the best person for this.

### Metadata
Note that this shows a kernel version not yet packaged in nixpkgs. That's mostly irrelevant, as this metadata reflects my running system and, since it's nixos-generate-config that needs to change to fix this, the problem actually occurred while booted from the stock 21.11 minimal installer image. I am not easily able to get back into that pre-install state to give the metadata pertaining to it.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.5, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by quinn-dougherty** [comment]
https://github.com/NixOS/nixos-hardware/pull/415 I just use it with this module, tho it took me a while to get as far as that.

**Comment by IreneKnapp** [comment]
yes, thank you. I am also the author of the acpilight lines in that module. however, I do think that since this is a topic that nixos-generate-config has a specific opinion on for other hardware, it should generate a correct configuration on this device rather than requiring the user to use external tooling.
